### DISCUSSION TITLE
How to transfer users from one instance of Appwrite to another

### DISCUSSION THREAD

**Discussion by ghost** [appwrite/appwrite#7762] [Q&A]
### 💭 Description

I am doing a complete refactor of my codebase. I'm figuring the easiest option will be to create a new instance of AppWrite on another server. I want to preserve my users. Is there any documentation or guidance on how to do a transfer of just the existing userbase? 

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by ItzNotABug** [comment]
You can dump all the user accounts into a JSON File via Users API & then create new accounts on the new Appwrite instance using relevant methods depending on the type of password hash used.

See - https://appwrite.io/docs/references/cloud/server-nodejs/users

**Reply by ghost** [reply]
I dumped all the users into a JSON, but it doesn't store the passwords. How do I transfer passwords with the user details? I can't even seem to dump hashed passwords

Also I'm reading the API link and it seems importing hashed passwords isn't a thing either?

**Comment by stnguyen90** [comment]
Another option is to use Appwrite's [migrations](https://appwrite.io/docs/advanced/migrations/self-hosted) feature.

**Reply by ghost** [reply]
It seems like that's only from self-hosted to the cloud, not from a self-hosted to a self-hosted?

**Reply by stephane2018** [reply]
i'm facing the same issue i have credentials error but the  credentials of the sender host is good no error but a have a `{
    "message": "Source Error: Failed to connect to mboasms.cm port 443 after 349 ms: Couldn't connect to server",
    "code": 400,
    "type": "migration_provider_error",
    "version": "1.5.4"
}` but  this url ok . 
Note: i'm the sender server version is 1.4.9 et the receiver server is 1.5.4
